modules/Task/components/TaskDetails/TaskDetails.tsx

```diff
@@ -80,19 +78,19 @@ const TaskDetails = ({ task, refetch }: IProps) => {
       </Group>
 
       <Text size="md" c="dimmed" mb={4}>
-        {task.description}
+        {task?.description}
       </Text>
 
       <Text size="md" c="dimmed" mb={4}>
         Priority:{" "}
-        <Badge color={getPriorityColor(task.priority)}>
-          {getPriority(task.priority)}
+        <Badge color={task && getPriorityColor(task.priority)}>
+          {task && getPriority(task.priority)}
```

Comment:
I don't think these checks are required. You are already checking if task exists or not when passing task from the TaskDetailsContainer. Please remove all of these task checks in this component.